src/main/java/com/wjoel/kafka_connect_psql_jsonb_sink/PsqlJsonbSinkTask.java

```diff
@@ -61,7 +61,7 @@ public void start(Map<String, String> props) {
                     "ALTER TABLE temp0 ALTER COLUMN " + valueColumn
                             + " TYPE json USING to_json(" + valueColumn + ")");
             String insertTempTableSql = "INSERT INTO " + table + " SELECT " + maybeKeyColumnComma
-                + valueColumn + "::jsonb FROM temp0";
+                + valueColumn + "::jsonb FROM temp0 ON CONFLICT DO NOTHING";
```

Comment:
Shouldn't it be `ON CONFLICT DO UPDATE` instead? That would make it an upsert. With `ON CONFLICT DO NOTHING` it will just leave the previous value (which may or may not be a duplicate).

Reply:
thats true... but how can you find which columns are key? I guess optionally we could delete the row and insert it again..

Reply:
Bah.... thats what the `keyColumn` is for, right?!

Reply:
Turns out that a unique index created on some fields of the jsonb cant be used for `ON CONFLICT` - it rather needs a unique index created on a whole column...

Reply:
We could do something like:
`ON CONFLICT (" + valueColumn + ") DO UPDATE SET " + valueColumn + " = EXCLUDED." + valueColumn + " WHERE " + keyColumn + " = EXCLUDED." + keyColumn;`
But the table should then have a unique index on the whole valueColumn. In our current case, we are indexing only some fields of the jsonb `valueColumn`.

Reply:
Im rather abandoning this change. I was under the assumption that duplicates may have been a problem of kafka connect but apparently our issue was something else.